src/ToggleButton/ToggleButtonOption.js

```diff
@@ -0,0 +1,326 @@
+      props.value = newValue;
+      this.props.onClick(props);
+    }
+  };
+
+  handleOptionClick = () => {
+    if (this.props.onClick && !this.props.children) {
+      this.props.onClick(this.props);
+    }
+  };
+
+  handleClick = event => {
+    this.setState({ open: true, anchorEl: event.currentTarget });
+  };
+
+  handleRequestClose = () => {
+    this.setState({ open: false });
+  };
+
+  render() {
+    const {
+      icon,
+      label,
+      classes,
+      color,
+      disabled,
+      divider,
+      toggle,
+      selected,
+      children,
+    } = this.props;
+
+    const buttonProps = {
+      className: classNames(classes.root, {
+        [classes.rootButton]: !toggle,
+        [classes.rootToggle]: toggle,
+        [classes.iconAndText]: icon && label,
+        [classes.buttonSelected]: selected && !toggle,
+        [classes.toggleDefault]: color === 'default' && selected && toggle && !disabled,
+        [classes.togglePrimary]: color === 'primary' && selected && toggle && !disabled,
+        [classes.toggleAccent]: color === 'accent' && selected && toggle && !disabled,
+        [classes.toggleContrast]: color === 'contrast' && selected && toggle && !disabled,
+        [classes.divided]: divider,
+      }),
+      onClick: this.handleOptionClick,
+      disableRipple: true,
+    };
+    const rootProps = {
+      className: classNames(classes.buttonBase, {
+        [classes.textSelected]: selected && !toggle,
+        [classes.button]: !toggle || disabled,
+        [classes.colorDefault]: color === 'default' && toggle && !disabled,
+        [classes.colorPrimary]: color === 'primary' && toggle && !disabled,
+        [classes.colorAccent]: color === 'accent' && toggle && !disabled,
+        [classes.colorContrast]: color === 'contrast' && toggle && !disabled,
+      }),
+    };
+
+    let option;
+    // If there are children available, make it a DropDown Menu.
+    if (!children) {
+      option = (
+        <ButtonBase {...buttonProps}>
+          <div {...rootProps}>
+            {icon}
+            {label}
+          </div>
+        </ButtonBase>
+      );
+    } else {
+      const items = children.map(child => {
+        return React.cloneElement(child, {
+          onClick: () => this.handleDropDownClick(child),
+        });
+      });
+      const dropButtonProps = {
+        'aria-owns': 'option-menu',
+        'aria-haspopup': true,
+        className: classNames(classes.root, classes.rootButton, {
+          [classes.iconAndText]: icon && label,
+          [classes.buttonSelected]: selected && !toggle,
+          [classes.divided]: divider,
+        }),
+        onClick: this.handleClick,
+        // centerRipple: true,
+      };
+      const arrowProp = {
+        style: {
+          // TODO: Move this to get style
+          transform: label && icon ? 'translate(50%, 50%)' : 'none',
+          color: selected ? fade(common.black, 0.54) : fade(common.black, 0.3),
+        },
+      };
+      const menuProps = {
```

Comment:
Please take full advantage of JSX so we write all our components the same way (reducing entropy).

Reply:
> It can be an instance property. But then, it feels wrong to have this property at all. We need to remove it.
Which property are you referring to?

> We use the value/onChange` couple as much as we can for controlling components. How doesn't that match what's store? 
Because what's stored in this property is the position of the selected item, not the value it contains.